src/java.base/share/classes/java/lang/reflect/code/type/MethodRef.java

```diff
@@ -57,12 +59,14 @@ public sealed interface MethodRef permits MethodRefImpl {
 
     FunctionType type();
 
-    // Conversions
+    // Resolutions and model access
 
     Executable resolveToMember(MethodHandles.Lookup l) throws ReflectiveOperationException;
 
     MethodHandle resolveToHandle(MethodHandles.Lookup l) throws ReflectiveOperationException;
 
+    Optional<CoreOps.FuncOp> codeModel(MethodHandles.Lookup l) throws ReflectiveOperationException;
```

Comment:
If the result is optional, do we also need the `throws` ?

Reply:
Yes, an empty optional means the method that is referenced has no code model. If there is an error attempting to determine that or obtain the code model then an exception is thrown e.g., lookup does not have the appropriate permissions to access the code model.